### DISCUSSION TITLE
Questions about the Google Colab Notebooks

### DISCUSSION THREAD

**Discussion by CatarinaNSSilva** [fishsizeproject/Course-MLforImageProcessing#3] [Q&A]
Hi all, 

Here you can ask any questions you might have about the content of the notebooks.

**Comment by silviammg86** [comment]
Hi, Silvia here from Brazil

I'd like to know if you will (or there is) a code in R available here, or should we adapt the code ourselves.
Thank you for the great course.

**Reply by CatarinaNSSilva** [reply]
Hi Silvia,

All the scripts in the course are in Python. I know that some packages we are using (such as OpenCV) are available in R, but I haven't used them for R, so I am not sure if there will be any compatibility issues if you adapt the course code to R. But please feel free to try it out! We would be really interested to know how it goes!

Cheers,

**Comment by CatarinaNSSilva** [comment]
Hi everyone,

This post is to let you know that I have now corrected the bug that @pour2west found in the notebook of today's session (good eye!).

When we plot the confusion matrix in notebook[ G-Practice4-Model-training.ipynb](https://colab.research.google.com/drive/1YgIrWWiaex8ObtAbAJOAXlRRGUbElkYj#scrollTo=q0JIiRnNZEme), the labels are now in the correct oder:

`axis_labels = ["perch", "striped bass", "trout"] `

Also, to answer you question @pour2west about what order should you list your labels when using your dataset - when we create a confusion matrix using the sklearn.metrics library:

```
from sklearn.metrics import confusion_matrix
confusion_matrix = confusion_matrix(df_confusion["expected"], df_confusion["predicted"])
```

It does shuffle the order of our entries - it puts labels in alphabetical order. So you just need to modify the line of code above with your labels in alphabetical order.

Would you like to share the results of your model with us? :)
